pkg/signature/kms/hashivault/client.go

```diff
@@ -220,17 +221,28 @@ func (h *hashivaultClient) fetchPublicKey(_ context.Context) (crypto.PublicKey,
 		return nil, fmt.Errorf("could not parse transit key keys data as map[string]interface{}")
 	}
 
-	publicKeyPem, ok := keyMap["public_key"]
+	publicKey, ok := keyMap["public_key"]
 	if !ok {
 		return nil, errors.New("failed to read transit key keys: corrupted response")
 	}
 
-	strPublicKeyPem, ok := publicKeyPem.(string)
+	strPublicKey, ok := publicKey.(string)
 	if !ok {
 		return nil, fmt.Errorf("could not parse public key pem as string")
 	}
+	// vault returns the key type in the "name" field
+	keyType := keyMap["name"]
+	if keyType == "ed25519" {
+		// vault returns ed25519 public keys as base64 encoding of
+		// the raw bytes of the key
+		decodedPublicKey, err := base64.StdEncoding.DecodeString(strPublicKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to base64 decode ed25519 public key: %s", err)
+		}
```

Comment:
perhaps worth adding a defensive check to ensure `len(decodedPublicKey) == ed25519.PublicKeySize`?

Reply:
Yes, agree it's worth adding a check here. Thanks!